cms/static/js/views/license-selector.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Provide license selector view for license models.
+ * These license selectors are used to set the license for
+ * new or existing courses and videos.
+ */
+define(["js/views/baseview", "underscore", "gettext", "js/models/license", "js/views/license-view"],
+    function(BaseView, _, gettext, LicenseModel, LicenseView) {
+
+        var LicenseSelector = BaseView.extend({
+            events: {
+                "click .license-button" : "onLicenseButtonClick",
+            },
+
+            initialize: function(options) {
+                this.template = this.loadTemplate("license-selector");
+                if (!this.model) {
+                    this.model = new LicenseModel();
+                }
+                else if (!(this.model instanceof LicenseModel) && this.model.kind) {
+                    this.model = new LicenseModel(this.model);
+                }
+                this.licenseView = new LicenseView({
+                    model: this.model
+                });
+
+                // Rerender on model change
+                this.listenTo(this.model, 'change', this.render);
+                this.render();
+            },
+
+            render: function() {
+                this.$el.html(this.template({
+                    license_data: this.model.get('kind'),
+                    license_preview: this.licenseView.renderLicense()
+                }));
+
+                this.$el.addClass('license-selector');
+
+                this.renderLicenseButtons();
+
+                return this;
+            },
+
+            renderLicenseButtons: function() {
+                var kind, $cc;
+                kind = this.model.get('kind');
+                $cc = this.$el.find('.selected-cc-license-options');
+
+                if (!kind || kind === "NONE" || kind === "ARR") {
+                    this.$el.find('.license-button[data-license="ARR"]').addClass('selected');
+                    this.$el.find('.license-button[data-license="CC"]').removeClass('selected');
+                    $cc.hide();
+                }
+                else {
+                    var attr = kind.split("-");
+                    this.$el.find('.license-button').removeClass('selected');
+                    for(i in attr) {
+                        this.$el.find('.license-button[data-license="' + attr[i] + '"]').addClass('selected');
```

Comment:
same here, this is necessary to show the license attributes in selected state when the license is set.